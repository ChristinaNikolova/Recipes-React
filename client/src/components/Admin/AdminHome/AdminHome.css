.bold-admin-dashboard {
    font-size: 20px;
    font-weight: bold;
}

.admin-hr-dashboard {
    margin: 20px;
}

.admin-ul {
    text-align: center;
}

.admin-li {
    display: flex;
    flex-direction: column;
    margin-bottom: 60px;
}

.admin-li-wrapper .custom-btn:first-of-type {
    margin-right: 20px;
}

.admin-li-title {
    color: #54BAB9;
    font-size: 30px;
    font-weight: bold;
    line-height: 1.5;
    letter-spacing: 1.5px;
    margin-bottom: 10px;
}