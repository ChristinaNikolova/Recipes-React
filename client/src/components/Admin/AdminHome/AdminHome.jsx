import { useEffect } from 'react';
import { Link } from 'react-router-dom';

import * as authService from '../../../services/authService.js';

import './AdminHome.css';

function AdminHome({ history }) {
    useEffect(() => {
        if (!authService.isAdmin()) {
            history.push('/');
            return;
        }
    }, [])

    return (
        <section id="admin-home" className="section">
            <h1 className="title text-center cursive-font-style">Welcome to Administration</h1>
            <hr className="admin-hr-dashboard" />
            <ul className="admin-ul">
                <li className="admin-li">
                    <div className="admin-li-title">
                        Categories
                    </div>
                    <div className="admin-li-wrapper">
                        <Link to="/admin/categories"><button className="custom-btn">All Categories</button></Link>
                        <Link to="/admin/categories/create"><button className="custom-btn">Create Category</button></Link>
                    </div>
                </li>
                <li className="admin-li">
                    <div className="admin-li-title">
                        Ingredients
                    </div>
                    <div className="admin-li-wrapper">
                        <Link to="/admin/ingredients"><button className="custom-btn">All Ingredients</button></Link>
                        <Link to="/admin/ingredients/create"><button className="custom-btn">Create Ingredient</button></Link>
                    </div>
                </li>
            </ul>
        </section>
    );
}

export default AdminHome;