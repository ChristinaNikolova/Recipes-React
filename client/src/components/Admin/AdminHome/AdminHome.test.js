import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import AdminHome from './AdminHome';

it('should render AdminHome component', () => {
    const wrapper = shallow(<AdminHome />);
    expect(toJson(wrapper)).toMatchSnapshot();
});