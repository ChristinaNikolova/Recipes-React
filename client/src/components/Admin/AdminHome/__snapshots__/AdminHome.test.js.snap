// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render AdminHome component 1`] = `
<div
  className="admin-home-wrapper"
>
  <h1
    className="text-center cursive-font-style p-1"
  >
    Welcome to Administration
  </h1>
  <hr
    className="hr-admin-dashboard"
  />
  <div>
    <ul
      className="custom-font-li"
    >
      <li
        className="bold-admin-dashboard m-1"
      >
        Categories
        <div
          className="mt-2 mb-2"
        >
          <Link
            to="/admin/categories"
          >
            <button
              className="btn btn-secondary mr-2 remove-font all-admin-area"
            >
              All Categories
            </button>
          </Link>
          <Link
            to="/admin/categories/create"
          >
            <button
              className="btn btn-primary mr-2 remove-font create-admin-area"
            >
              Create Category
            </button>
          </Link>
        </div>
      </li>
      <li
        className="bold-admin-dashboard m-1"
      >
        Ingredients
        <div
          className="mt-2 mb-2"
        >
          <Link
            to="/admin/ingredients"
          >
            <button
              className="btn btn-secondary mr-2 remove-font all-admin-area"
            >
              All Ingredients
            </button>
          </Link>
          <Link
            to="/admin/ingredients/create"
          >
            <button
              className="btn btn-primary mr-2 remove-font create-admin-area"
            >
              Create Ingredient
            </button>
          </Link>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
